### PR TITLE
Add Language: Neos:Fusion

### PR DESCRIPTION
**Pull request by amtee**
We used highlight.js  to build a syntax hightlighting for [Fusion](https://neos.readthedocs.io/en/stable/CreatingASite/Fusion/InsideFusion.html)

Now giving it back.

### CODE REVIEW

File: src/languages/fusion.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Language: Fusion
+TODO: add categories
+ */
+
+function (hljs) {
+  var KEYWORDS = {
+    keyword: 'prototype include namespace @process @position @ignoreProperties @context @allowEmpty @if @cache',
+    literal: 'true false null'
+  };
+  // Already added for future implementation, does not work atm
```
Comment (marcoscaceres) [inline src/languages/fusion.js]: Nit: to what does this comment apply to?

**Comment by amtee** [comment]
I merged master into the PR and changed the comment you were asking about @marcoscaceres

**Comment by marcoscaceres** [comment]
The failing test on Travis appear to be related to fusion. Could you use `nvm` to replicate the node version and see why the test is failing? Or maybe it's obvious from looking at the travis logs?

**Comment by amtee** [comment]
Ok, still cannot get this to work. Put some hours of effort in it, but that's it for now.

When I try the developer.html auto highlighting everythings fine. But `npm run test`  breaks now when livescript is beeing detected as scala.

Maybe some of the experienced maintainers wants to keep up on this.

Cheers

**Comment by marcoscaceres** [comment]
Currently working our way through the backlog - but will try to get to this. Other maintainers, help appreciated! 🙏

**Comment by egor-rogov** [comment]
Looks like two problems are here.

The first one is the file `test/detect/livescript/default.expect.txt`. It confuses the testing engine, which tries to autodetect this file instead of `default.txt` and eventually gets **scala**. Just remove this file (and `test/detect/fusion/default.expect.txt` too).

And this leads to the second problem: now **livescript** autodetects as **fusion**. This is reproducible in developer.html. Unfortunately I have no idea what to do about it  as I'm not familiar with both of the languages.

**Comment by yyyc514** [comment]
@amtee 

**First of all, truly sorry for the long delay in anyone responding to this PR.**

Thank you so much for desiring to contribute and writing a language grammar.  Unfortunately, we currently are no longer accepting new languages into the core repository.  The existing maintainers simply don't have time to maintain new languages (fixing bugs, dealing with issues, etc).  **We'd still love to host your language as a 3rd party module though** - if you're willing to help maintain it.  

It's easy to turn your language into a module that anyone who likes can easily integrate with Highlight.js using a single line of code:

```html
<script src="/path/to/highlight.min.js"></script>
<!-- they'd simply add your language modules -->
<script src="/path/to/highlightjs-neos-fusion.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
```

It's also possible for users to rebuild the `highlight.js` library to include specific 3rd party modules, all in a single file (vs loading a bunch of 3rd party languages one at a time).

If you'd like to become the maintainer for this language then we're happy to create a new repository in the `highlightjs` organization and host the repository there.  That assumes you'd be open to reviewing issues and fixing bugs as they arise.  You're also welcome to host the repo yourself and we'll link to it from our README.

There isn't much in the way of how to docs on this yet, but there are sample repositories and it's honestly pretty easy.  We're still fleshing out an [official "spec" for 3rd party modules](https://github.com/highlightjs/highlight.js/issues/2328), but even after that's done it should require minimal changes from the examples we have already.  IE, if you wanted to get started now there isn't any big reason to wait.

My own `robots.txt` grammar repository is a great, simple example:

https://github.com/highlightjs/highlightjs-robots-txt

If you have any questions feel free to ask.

**Comment by yyyc514** [comment]
> And this leads to the second problem: now livescript autodetects as fusion. This is reproducible in developer.html. Unfortunately I have no idea what to do about it as I'm not familiar with both of the languages.

@amtee If you're wanting to do the work to turn this into a 3rd party grammar module and become the maintainer I'd be happy to take a closer look at any remaining test conflicts and see if we could get them sorted out.

The good news is the test conflicts won't prevent you from publishing the module repository and getting things going initially.

**Comment by yyyc514** [comment]
**Closing for inactivity;** (also finally cleaning all these old PRs out for the new year).

As mentioned above if you're interested in becoming the maintainer of a 3rd party module, please respond [whenever you find time] and we'd be happy to help you get started.

**Pull request closed without merging**